add named parameters to mobile module

ann-ss opened the pull request:
torch::jit::mobile::Module does not currently support accessing parameters via their attribute names, but torch::jit::Module does. This diff adds an equivalent functionality to mobile::Module.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #41403 implement lite parameter serializer
* **#41376 add named parameters to mobile module**

Differential Revision: [D22609142](https://our.internmc.facebook.com/intern/diff/D22609142)

ann-ss committed:
add named parameters to mobile module

[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"




[ghstack-poisoned]

iseeyuan reviewed (approved):
Awesome work. Thanks!

ann-ss committed:
Update on "add named parameters to mobile module"

torch::jit::mobile::Module does not currently support accessing parameters via their attribute names, but torch::jit::Module does. This diff adds an equivalent functionality to mobile::Module.


Differential Revision: [D22609142](https://our.internmc.facebook.com/intern/diff/D22609142)

[ghstack-poisoned]

ann-ss committed:
Update on "add named parameters to mobile module"

torch::jit::mobile::Module does not currently support accessing parameters via their attribute names, but torch::jit::Module does. This diff adds an equivalent functionality to mobile::Module.


Differential Revision: [D22609142](https://our.internmc.facebook.com/intern/diff/D22609142)

[ghstack-poisoned]
